cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -0,0 +1,172 @@
+import os
+
+import yaml
+from aws_cdk import aws_iam as iam
+from aws_cdk import aws_imagebuilder as imagebuilder
+from aws_cdk import aws_ssm as ssm
+from aws_cdk import core
+
+from common.utils import load_yaml
+from pcluster.imagebuilder import ImageBuilder
+
+
+class ImageBuilderStack(core.Stack):
+    """Create the Stack for imagebuilder."""
+
+    def __init__(self, scope: core.Construct, construct_id: str, imagebuild: ImageBuilder, **kwargs) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+        self.imagebuild = imagebuild
+        image_config = imagebuild.config.image_config
+        build_config = imagebuild.config.build_config
+        dev_settings_config = imagebuild.config.dev_settings_config
+
+        # TODO: use attributes from imagebuilder config instead of using these static variables.
+        core.CfnParameter(self, "EnableNvidia", type="String", default="false", description="EnableNvidia")
+        core.CfnParameter(self, "EnableDCV", type="String", default="false", description="EnableDCV")
+
+        # TODO: get archecture by ami id
+        architecture = "x86_64"
+        instance_types = ["p2.xlarge", "c5.xlarge"] if architecture == "x86_64" else ["m6g.xlarge"]
+
+        # Setup ImageBuilder Resources
+        # InstanceRole
+        if build_config.instance_role:
+            managed_policy_arns = [build_config.instance_role]
+        else:
+            managed_policy_arns = [
+                core.Fn.sub("arn:${AWS::Partition}:iam::aws:policy/AmazonSSMManagedInstanceCore"),
+                core.Fn.sub("arn:${AWS::Partition}:iam::aws:policy/EC2InstanceProfileForImageBuilder"),
+            ]
+        instancerole = iam.CfnRole(
+            self,
+            id="InstanceRole",
+            managed_policy_arns=managed_policy_arns,
+            assume_role_policy_document={
+                "Statement": {
+                    "Action": "sts:AssumeRole",
+                    "Effect": "Allow",
+                    "Principal": {"Service": "ec2.amazonaws.com"},
+                },
+                "Version": "2012-10-17",
+            },
+            path="/executionServiceEC2Role/",
+        )
+        instancerole.add_metadata("Comment", "Role to be used by instance during image build.")
+
+        # InstanceProfile
+        iam.CfnInstanceProfile(
+            self, id="InstanceProfile", path="/executionServiceEC2Role/", roles=[core.Fn.ref("InstanceRole")]
+        )
+
+        # InfrastructureConfiguration
+        imagebuilder.CfnInfrastructureConfiguration(
+            self,
+            id="PClusterImageInfrastructureConfiguration",
+            name="PCluster-Image-Infrastructure-Configuration",
```

Comment:
name should be unique, we can use the utility function `generate_random_name_with_prefix`

Reply:
Done